wiring/inc/spark_wiring_system.h

```diff
@@ -106,6 +108,167 @@ inline SleepResult::SleepResult(WakeupReason r, system_error_t e, pin_t p)
       pin_(p) {
 }
 
+enum class SystemSleepWakeupReason: uint16_t {
+    UNKNOWN = HAL_WAKEUP_SOURCE_TYPE_UNKNOWN,
+    BY_GPIO = HAL_WAKEUP_SOURCE_TYPE_GPIO,
+    BY_ADC = HAL_WAKEUP_SOURCE_TYPE_ADC,
+    BY_DAC = HAL_WAKEUP_SOURCE_TYPE_DAC,
+    BY_RTC = HAL_WAKEUP_SOURCE_TYPE_RTC,
+    BY_LPCOMP = HAL_WAKEUP_SOURCE_TYPE_LPCOMP,
+    BY_UART = HAL_WAKEUP_SOURCE_TYPE_UART,
+    BY_I2C = HAL_WAKEUP_SOURCE_TYPE_I2C,
+    BY_SPI = HAL_WAKEUP_SOURCE_TYPE_SPI,
+    BY_TIMER = HAL_WAKEUP_SOURCE_TYPE_TIMER,
+    BY_CAN = HAL_WAKEUP_SOURCE_TYPE_CAN,
+    BY_USB = HAL_WAKEUP_SOURCE_TYPE_USB,
+    BY_BLE = HAL_WAKEUP_SOURCE_TYPE_BLE,
+    BY_NFC = HAL_WAKEUP_SOURCE_TYPE_NFC,
+    BY_NETWORK = HAL_WAKEUP_SOURCE_TYPE_NETWORK,
+};
+
+class SystemSleepResult {
+public:
+    SystemSleepResult()
+            : wakeupSource_(nullptr),
+              error_(SYSTEM_ERROR_NONE) {
+    }
+
+    SystemSleepResult(SleepResult r)
```

Comment:
If construct result by this way, calling methods of `SystemSleepResult` may return incorrect result information, because the `wakeupSource_` is nullptred.